src/App.js

```diff
@@ -1,16 +1,53 @@
-import React from 'react';
+// import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
+const CHATLOG = chatMessages;
 
 const App = () => {
+  // wave 3
+  const chatLogCopy = CHATLOG.map((message) => {
+    return {
+      id: message.id,
+      sender: message.sender,
+      body: message.body,
+      timeStamp: message.timeStamp,
+      liked: message.liked,
+    };
+  });
+  const [chatLogState, setChatLogState] = useState(chatLogCopy);
+
+  const toggleLike = (id) => {
+    const msgs = chatLogState.map((msg) => {
+      if (msg.id === id) {
+        msg.liked = !msg.liked;
+        countLikes(msg);
+      }
+      return msg;
+    });
+    setChatLogState(msgs);
+  };
+
+  let [totalLikes, setTotalLikes] = useState(0);
```

Comment:
Make sure all your React state variables are created as `const` variables. This can be a little unintuitive since the value of `totalLikes` is clearly going to change.

It's important to note that when a component re-renders, it creates a brand new scope with brand new variable declarations. So, the `useState()` hook works by setting the initial value the state variable gets created with during the next re-render. So, under the hood, we aren't changing the value of `totalLikes` from an old value to a new value when we re-render. We are scrapping the old variable completely, and starting fresh with a new state variable declaration with a new initial value.

The other reason we want to declare state variables as `const` is because this prevents direct manipulation of the state variable. As you know, we should only update state variables using their setter functions. So, declaring the state variable as `const` enforces this practice.